go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -154,6 +157,14 @@ func NewTabletServer(name string, config *tabletenv.TabletConfig, topoServer *to
 	tsv.txThrottler = txthrottler.NewTxThrottler(tsv.config, topoServer)
 	tsv.te = NewTxEngine(tsv)
 	tsv.messager = messager.NewEngine(tsv, tsv.se, tsv.vstreamer)
+	tsv.lagThrottler = throttle.NewThrottler(tsv, topoServer,
+		func() topodatapb.TabletType {
+			if tsv.sm == nil {
+				return topodatapb.TabletType_UNKNOWN
+			}
+			return tsv.sm.Target().TabletType
+		},
```

Comment:
nit: this function can be moved to throttler code and tsv.sm can be passed into that function

Reply:
I felt like it would add a bit too much dependency (also `import`-order -wise) if throttler were to use state manager. What do you think

Reply:
the dependency is already there via closure. We can have it explicitly as well. You can take a call.

Reply:
Thank you! Logic-wise the dependency is there; `golang` `import`-wise, it's less of a dependency (I guess idiomatically I should have passed an `interface` rather than a function though)